src/tysts.ts

```diff
@@ -275,7 +275,6 @@ import { FromCoercer, coercer } from ".";
     const coerceUser = coercer<User>(($) => ({
       name: String,
       credentials: $.Union(
-        // @ts-expect-error: Object unions aren't suported right now
```

Comment:
The problem wasn't that the types wouldn't accept it. It's the opposite — I needed to restrict `$.Union` function from accepting objects as they are very tricky to coerce. See my comment: https://github.com/kossnocorp/ofcoerce/pull/4#issuecomment-2170947739

Reply:
Hmmm you can probably restrict it to primitives + coerses, since those can be distinguished with their symbol both in type and in runtime. I think that is preferable in general

Reply:
Yeah, I think this is the way. I want to approach it gradually, though. Everything at once doesn't really work for me with complex generics.